### PR TITLE
WIP first working netcoreapp3.1 migration

### PR DESCRIPTION
**Pull request by MarkZither**
WIP migration to .netcoreapp3.1, so bits to finish around the routing and some bits to revert.

 * Update the project file and remove/add relevant package references
 * Change BaGet.Core.Server to use Microsoft.NET.Sdk.Web and target netcoreapp3.1 for MVC references
 * Change from IHostingEnvironment to IWebHostEnvironment
 Fix a duplicate route that is throwing and exception for SymbolDownloadRouteName as also done in wagnerhsu's fork
 * Add .AddNewtonsoftJson(options => { options.SerializerSettings.DateTimeZoneHandling = DateTimeZoneHandling.Utc; });
* Migrate from deprecated UseMvc to UseRouting and UseEndpoints (having issues finding the controllers in BaGet.Core.Server. Checking Share controllers, views, Razor Pages and more with Application Parts and .NET core 3 route failure with separate projects

Addresses https://github.com/loic-sharma/BaGet/issues/439

### CODE REVIEW

File: src/BaGet.Core.Server/BaGet.Core.Server.csproj [hunk 1]
```diff
@@ -1,16 +1,13 @@
-﻿<Project Sdk="Microsoft.NET.Sdk">
+<Project Sdk="Microsoft.NET.Sdk.Web">
```
Comment (loic-sharma) [inline src/BaGet.Core.Server/BaGet.Core.Server.csproj]: Would it be better to use `<FrameworkReference Include="Microsoft.AspNetCore.App" />` instead? With the framework reference you won't need an entry point in `OperationCancelledMiddleware.cs`

Comment (loic-sharma) [inline src/BaGet.Core.Server/BaGet.Core.Server.csproj]: Would it be better to use `<FrameworkReference Include="Microsoft.AspNetCore.App" />` instead? With the framework reference you won't need an entry point in `OperationCancelledMiddleware.cs`

File: src/BaGet.Core.Server/Extensions/IRouteBuilderExtensions.cs [hunk 2]
```diff
@@ -6,40 +7,25 @@ namespace BaGet.Extensions
 {
     public static class IRouteBuilderExtensions
     {
-        public static IRouteBuilder MapServiceIndexRoutes(this IRouteBuilder routes)
+        public static IEndpointRouteBuilder MapServiceIndexRoutes(this IEndpointRouteBuilder endpoints)
```
Comment (loic-sharma) [inline src/BaGet.Core.Server/Extensions/IRouteBuilderExtensions.cs]: Is it possible to port to endpoint routing in a separate pull request? I'd like to minimize the changes as the risk of breaking changes is already quite high.

**Comment by loic-sharma** [comment]
> having issues finding the controllers in BaGet.Core.Server. Checking Share controllers, views, Razor Pages and more with Application Parts and .NET core 3 route failure with separate projects

I was able to get routing to work by removing `Async` suffixes from the action names in my routes. See this comment: https://github.com/loic-sharma/BaGet/issues/439#issuecomment-576043268. My link generation is still broken though, I'm still investigating that

File: src/BaGet.Core.Server/Extensions/IRouteBuilderExtensions.cs (hunk 2 above)
Comment (MarkZither) [reply src/BaGet.Core.Server/Extensions/IRouteBuilderExtensions.cs]: yes, i'll update the PR shortly

**Comment by loic-sharma** [comment]
Hi @MarkZither, I merged the update to ASP.NET Core 3.0 (https://github.com/loic-sharma/BaGet/pull/444) and opened a new pull request for ASP.NET Core 3.1 (https://github.com/loic-sharma/BaGet/pull/446). I fixed the EF Core 3.1 warnings like `no value comparer`.

I'm currently investigating exceptions thrown by the ASP.NET Core 3.1 SPA services... It seems proxying is a little flaky, but I'm not sure why.

**Comment by loic-sharma** [comment]
Hey @MarkZither, I merged your changes as part of https://github.com/loic-sharma/BaGet/pull/446. Thank you for starting this project!

**Pull request closed without merging**